User guide for logging framework in Tinkerbell

Cbkhare opened the pull request:
Signed-off-by: cbkhare <Chitrabasukhare89@gmail.com>

## Description

This is the PR for documentation of Tinkerbell Centralized logging. This is with reference to following PRs. 

boots PR: https://github.com/tinkerbell/boots/pull/73
Tink PR: https://github.com/tinkerbell/tink/pull/267
OSIE Pr: https://github.com/tinkerbell/osie/pull/104
RFD: https://github.com/tinkerbell/proposals/pull/16

gianarb commented on content/docs/troubleshooting/logging.md:
```diff
@@ -0,0 +1,43 @@
+- tink-worker
+- action containers
+
+
+Logging is crucial for any product or feature to ensure debugging and bug-free implementation. Logs of these services can be obtained using `docker logs <service name>`. For tink-worker and action containers, the steps are same and many a times it would happend that service containers has exited once the execution is completed leaving the user with an overhead of first searching the dead containe ...
```
`bug-free implementation`? :) nop, let's leave it out please. That's a dream!

gianarb replied:
Let's be very clear.

Tinkerbell leverage Docker in two different ways. First, you can run the above services as a container. Second, every action is a container by itself.

Cbkhare replied:
you are suggesting to segregate the explanation here in terms of docker usage for provisioner and worker?

gianarb commented on content/docs/troubleshooting/logging.md:
```diff
@@ -0,0 +1,43 @@
+
+## Implementation details
+
+Centralized logging only captures docker or microservices logs. It leverages the [`logging driver`](https://docs.docker.com/config/containers/logging/configure/) and `logging option` parameter of the docker cli. Currently, we are supporting `server_address` and `tag` log-opts but in next releases, we will support other log-opts as well.
```
You can centralize logs configuring the docker daemon with the right logging driver.

gianarb commented on content/docs/troubleshooting/logging.md:
```diff
@@ -0,0 +1,43 @@
+
+For the docker container configuration we have used
+- `log-driver`: `syslog`
+- `server-address`: `tcp://<tinkerbell_ip>:514`
```
`provisioner_ip`

Cbkhare replied:
okay, I will change it.ideally, it should have been the log_server_address_ip.

gianarb replied:
Even better!

Cbkhare replied:
done

gianarb commented:
Let's call this document something like: "Logging in Tinkerbell."

The first section should explain more clearly the relation between the service tinkerbell is made of and docker. Running boots, tink-server inside a container is only one way to do it. Let's be clear about it.

Do not mention the action log at the beginning but only in their own section, because I think merging services and actions together creates confusions. We have to keep separated system needs (what is required to run tinkerbell) and what tinkerbell does.

In second, when those 2 sections (first about services, second about actions) are well described let's have a third one when you mention that logs can be centralized configuring docker logging driver.

At this point I would like to close this document as it is. Without giving a concrete implementation.

You can create a new document called: "Centralize logs with syslog".
This document will be attached as subpage in the `Vagrant Setup` menu, and you should describe how to setup syslog server in the vagrant provisioner and how to centralize and read logs from it.

Does it sound clear?
Thanks

gianarb commented:
I don't think this work reference the PRs you linked btw.

Cbkhare commented:
> Let's call this document something like: "Logging in Tinkerbell."
> 
> The first section should explain more clearly the relation between the service Tinkerbell is made of and docker. Running boots, tink-server inside a container is only one way to do it. Let's be clear about it.
> 
> Do not mention the action log at the beginning but only in their own section, because I think merging services and actions together create confusions. We have to keep separated system needs (what is required to run Tinkerbell) and what Tinkerbell does.
> 
> In second, when those 2 sections (first about services, second about actions) are well described let's have a third one when you mention that logs can be centralized configuring docker logging driver.
> 
> At this point, I would like to close this document as it is. Without a concrete implementation.
> 
> You can create a new document called: "Centralize logs with syslog".
> This document will be attached as a subpage in the `Vagrant Setup` menu, and you should describe how to setup a Syslog server in the vagrant provisioner and how to centralize and read logs from it.
> 
> Does it sound clear?
> Thanks

@gianarb This implementation is independent of the type of deployment. As of now, I have completed the testing with Tinkerbell and also completed plug and play experiment. Next, I will try it with terraform setup and also with sandbox (if it's ready). 
Moreover, I have a preference to merge it only after the testing with all form of deployment is completed.

Cbkhare commented:
@mmlb I have addressed your comments on this https://github.com/tinkerbell/proposals/pull/16 and have added them in this PR. 

@gianarb I have also addressed your comments. 

Currently, I am doing some testing with `rsyslog` and `fluent`. I will push the suggested changes along with details of `Plug and Play` model. 

I am not sure, maybe can decline this PR https://github.com/tinkerbell/proposals/pull/16. Or I think, having this file can be helpful to understand RFD 003. We can update RFD PR after this PR is merged.

Cbkhare commented:
On my PRs for the feature of Centralised logging i have added the support where user can use the Tinkerbell with existing logging system i.e with docker logs <service-name> . And, at the same time, if user wishes, he/she can also use the centralized logging system as per the choice of his/her log driver by updating some set of env variables. Also as a part of the implementation, we will give the base platform and configuration to run centralized logging with rsyslog.
 I will update the doc https://github.com/tinkerbell/tinkerbell.org/pull/135 in some time, but PR for tink/boots/osie is ready.
